q2_fmt/_peds.py

```diff
@@ -486,8 +507,58 @@ def _create_masking(time_metadata, donor_df, recip_df, reference_column):
 
 
 def _mask_recipient(donor_mask, recip_df):
-    recip_mask = donor_mask & recip_df
-    return recip_mask
+    maskedrecip = donor_mask & recip_df
+    return maskedrecip
+
+
+def sample_pprs(table: pd.DataFrame, metadata: qiime2.Metadata,
+                time_column: str, baseline_timepoint: str, subject_column: str,
+                filter_missing_references: bool = False,
+                drop_incomplete_subjects: bool = False,
+                drop_incomplete_timepoints: list = None) -> (pd.DataFrame):
+    ids_with_data = table.index
+    metadata = metadata.filter_ids(ids_to_keep=ids_with_data)
+    column_properties = metadata.columns
+    # TODO: Make incomplete samples possible move this to heatmap
+    metadata = metadata.to_dataframe()
+    if drop_incomplete_timepoints is not None:
+        metadata = _drop_incomplete_timepoints(metadata, time_column,
+                                               drop_incomplete_timepoints)
+        table.filter(items=metadata.index)
+    num_timepoints = _check_for_time_column(metadata, time_column)
+    _check_column_type(column_properties, "time",
+                       time_column, "numeric")
+
+    used_references =\
+        _get_to_baseline_ref(time_col=metadata[time_column],
+                             baseline_timepoint=baseline_timepoint,
+                             time_column=time_column,
+                             subject_column=subject_column,
+                             metadata=Metadata(metadata))
+
+    subject_series = _check_subject_column(metadata, subject_column)
+    _check_column_type(column_properties, "subject",
+                       subject_column, "categorical")
+    _check_duplicate_subject_timepoint(subject_series, metadata,
+                                       subject_column, time_column)
+    # return things that should be removed
+    metadata, used_references = \
+        _check_subjects_in_all_timepoints(subject_series, num_timepoints,
+                                          drop_incomplete_subjects, metadata,
+                                          subject_column, used_references)
+
+    peds_df = pd.DataFrame(columns=['id', 'measure',
+                                    'transfered_baseline_features',
+                                    'total_baseline_features', 'baseline',
+                                    'subject', 'group'])
+    baseline_metadata = metadata.join(used_references)
+    peds_df = _compute_peds(peds_df=peds_df, peds_type="PPRS",
```

Comment:
okay so making sure i understand this - the actual computation is the same here, the only difference is what you're comparing, which is the baseline features that were retained post-FMT against the total baseline features (i.e. everything the recipient started with pre-FMT)?

Reply:
Yeah exactly!